src/main/java/com/mineboundteam/minebound/magic/UtilitySpells/FireUtilitySpell.java

```diff
@@ -0,0 +1,186 @@
+            List<FireUtilitySpell> equippedSpells = getEquippedSpellItemsOfType(FireUtilitySpell.class, player);
+            if (equippedSpells.size() > 0) {
+                event.setCanceled(true);
+                player.clearFire();
+            }
+        }
+    }
+
+    @Override
+    public void appendHoverText(ItemStack pStack, @Nullable Level pLevel, List<Component> pTooltipComponents, TooltipFlag pIsAdvanced) {
+        super.appendHoverText(pStack, pLevel, pTooltipComponents, pIsAdvanced);
+        pTooltipComponents.add(new TextComponent("While equipped in a ").withStyle(ChatFormatting.GRAY)
+                .append(new TextComponent("utility slot").withStyle(ChatFormatting.DARK_PURPLE))
+                .append(":"));
+        double rateInSeconds = damageRate / 20d;
+        pTooltipComponents.add(new TextComponent("  - To all entities in an ").withStyle(ChatFormatting.GRAY)
+                .append(new TextComponent("AOE of " + aoeRange + " blocks").withStyle(ChatFormatting.DARK_GREEN))
+                .append(", deals ")
+                .append(new TextComponent(new DecimalFormat("0.#").format(damage) + " heart" + (damage != 1 ? "s" : "") + " of fire damage").withStyle(ChatFormatting.RED))
+                .append(" every ")
+                .append(new TextComponent(new DecimalFormat("0.#").format(rateInSeconds) + " second" + (rateInSeconds > 1 ? "s" : "")).withStyle(ChatFormatting.DARK_GREEN)));
+        pTooltipComponents.add(new TextComponent("  - Gives fire immunity").withStyle(ChatFormatting.GRAY));
+        pTooltipComponents.add(new TextComponent("Multiple copies stack to increase the ").withStyle(ChatFormatting.GRAY)
+                .append(new TextComponent("range").withStyle(ChatFormatting.DARK_GREEN))
+                .append(", ")
+                .append(new TextComponent("damage").withStyle(ChatFormatting.RED))
+                .append(", and ")
+                .append(new TextComponent("Mana cost").withStyle(manaColorStyle)));
+        pTooltipComponents.add(new TextComponent("Costs ").withStyle(ChatFormatting.GRAY)
+                .append(new TextComponent(manaCost + " Mana").withStyle(manaColorStyle)).append(" per entity damaged"));
+
+        LocalPlayer player = Minecraft.getInstance().player;
+        if (player != null) {
+            MutableComponent active = new TextComponent("Fire AOE is currently ").withStyle(ChatFormatting.UNDERLINE).withStyle(ChatFormatting.BOLD);
+            Optional<PlayerUtilityToggleProvider.UtilityToggle> toggle = player.getCapability(PlayerUtilityToggleProvider.UTILITY_TOGGLE).resolve();
+            if (toggle.isPresent() && toggle.get().fire)
+                pTooltipComponents.add(active.append(new TextComponent("ON").withStyle(ChatFormatting.GREEN)));
+            else
+                pTooltipComponents.add(active.append(new TextComponent("OFF").withStyle(ChatFormatting.RED)));
+        }
+    }
+
+    public static class FireUtilitySpellConfig implements IConfig {
```

Comment:
Since it gives an effect we should probably also reduce mana cap by some amount.

Reply:
Does 50 for all tiers sound good?

Reply:
50 is a lot compared to some of the others; I'd say 20 or 25, unless we want to rebalance some of the other values (which we might want to do anyway)

Reply:
I'll do 25 in that case and we can adjust it later if we want.

Reply:
Done